test/performance/odp_bench_event_vector.c

```diff
@@ -0,0 +1,760 @@
+	BENCH_INFO(event_vector_size_set, create_event_vectors, free_event_vectors, NULL),
+	BENCH_INFO(event_vector_user_flag, create_event_vectors, free_event_vectors, NULL),
+	BENCH_INFO(event_vector_user_flag_set, create_event_vectors, free_event_vectors, NULL),
+};
+
+ODP_STATIC_ASSERT(ODPH_ARRAY_SIZE(test_suite) < TEST_MAX_BENCH,
+		  "Result array is too small to hold all the results");
+
+static int create_pools(void)
+{
+	odp_pool_capability_t capa;
+	odp_pool_param_t evv_pool_params;
+	odp_pool_param_t buf_pool_params;
+
+	if (odp_pool_capability(&capa)) {
+		ODPH_ERR("Error: unable to query pool capability\n");
+		return -1;
+	}
+
+	if (capa.event_vector.max_pools == 0) {
+		ODPH_ERR("Error: event vector pools not supported\n");
+		return -1;
+	}
+
+	if (capa.event_vector.max_num && capa.event_vector.max_num < TEST_REPEAT_COUNT) {
+		ODPH_ERR("Error: pool size not supported (max %" PRIu32 ")\n",
+			 capa.event_vector.max_num);
+		return -1;
+	} else if (gbl_args->appl.cache_size > (int)capa.event_vector.max_cache_size) {
+		ODPH_ERR("Error: cache size not supported (max %" PRIu32 ")\n",
+			 capa.event_vector.max_cache_size);
+		return -1;
+	}
+
+	gbl_args->buf_size = TEST_BUF_SIZE;
+
+	if (capa.event_vector.max_size && capa.event_vector.max_size < TEST_EVV_SIZE)
+		gbl_args->evv_size = capa.event_vector.max_size;
+
+	if (gbl_args->evv_size > capa.event_vector.max_size ||
+	    gbl_args->evv_size > TEST_MAX_EVV_SIZE) {
+		ODPH_ERR("ERR: event vector size not supported (max %" PRIu32 ")\n",
+			 capa.event_vector.max_size < TEST_MAX_EVV_SIZE ?
+			 capa.event_vector.max_size : TEST_MAX_EVV_SIZE);
+		return -1;
+	}
+
+	gbl_args->evv_uarea_size = TEST_UAREA_SIZE < capa.event_vector.max_uarea_size ?
+				TEST_UAREA_SIZE : capa.event_vector.max_uarea_size;
+
+	gbl_args->buf_uarea_size = TEST_UAREA_SIZE < capa.buf.max_uarea_size ?
+				TEST_UAREA_SIZE : capa.buf.max_uarea_size;
+
+	print_info();
+
+	/* Create event vector pool */
+	odp_pool_param_init(&evv_pool_params);
+	evv_pool_params.event_vector.max_size = gbl_args->evv_size;
+	evv_pool_params.event_vector.num = TEST_REPEAT_COUNT;
+	evv_pool_params.event_vector.uarea_size = gbl_args->evv_uarea_size;
+	if (gbl_args->appl.cache_size >= 0)
+		evv_pool_params.event_vector.cache_size = gbl_args->appl.cache_size;
+	evv_pool_params.type = ODP_POOL_EVENT_VECTOR;
+
+	gbl_args->evv_pool = odp_pool_create("microbench event vector", &evv_pool_params);
+	if (gbl_args->evv_pool == ODP_POOL_INVALID) {
+		ODPH_ERR("Error: pool create failed\n");
+		return -1;
+	}
+
+	odp_pool_print(gbl_args->evv_pool);
+
+	/* Create buffer pool */
+	odp_pool_param_init(&buf_pool_params);
+	buf_pool_params.buf.size = gbl_args->buf_size;
+	buf_pool_params.buf.num = TEST_REPEAT_COUNT * gbl_args->evv_size;
+	buf_pool_params.buf.uarea_size = gbl_args->buf_uarea_size;
+	if (gbl_args->appl.cache_size >= 0)
+		buf_pool_params.buf.cache_size = gbl_args->appl.cache_size;
```

Comment:
Buffer pool cache size does not matter for this test, so can be left to default.